src/components/SidebarState.vue

```diff
@@ -43,19 +43,37 @@ export default Vue.extend({
     },
     methods: {
         section() {
-            if (this.isClosed) {
+            if (!this.sidebarOpen) {
                 return '';
             }
 
             let section = this.sidebarSection;
-            let isChannel = this.$state.getActiveBuffer().isChannel();
+            let buffer = this.$state.getActiveBuffer();
 
-            if (section === 'settings' && isChannel) {
-                return 'settings';
-            } else if (section === 'user' && this.sidebarUser && isChannel) {
-                return 'user';
-            } else if (section === 'nicklist' && isChannel) {
+            if (buffer.isQuery()) {
+                let user = this.$state.getUser(buffer.getNetwork().id, buffer.name);
+                if (user) {
+                    this.sidebarUser = user;
```

Comment:
`section()` is used for getting the current section, yet this has a side effect of setting the user. This would be better in a computed property that the template reads from instead

Reply:
I've made better comments for why this side effect exists.

Basically its so the sidebar will show a user when switching to pm without changing the state from nicklist so when you return to a channel with a nicklist it will be shown